cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/hdfs/HdfsDelegationTokenRefresherConfig.java

```diff
@@ -0,0 +1,55 @@
+package com.linkedin.openhouse.cluster.storage.hdfs;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableScheduling;
+
+/**
+ * Configuration class to conditionally enable/disable creation of {@link
+ * HdfsDelegationTokenRefresher} bean and enable delegation token refresh scheduling.
+ */
+@Slf4j
+@Configuration
+@EnableScheduling
+public class HdfsDelegationTokenRefresherConfig {
+
+  @Autowired private HdfsStorage hdfsStorage;
+
+  private static final String HDFS_TOKEN_REFRESH_ENABLED = "token.refresh.enabled";
+
+  /**
+   * Conditionally provide the HdfsDelegationTokenRefresher bean if the parameter for token refresh
+   * is enabled in the HdfsStorage properties. The relevant configuration in the cluster YAML file
+   * is as follows:
+   *
+   * <pre>
+   * cluster:
+   *   storages:
+   *     hdfs:
+   *       parameter:
+   *         token.refresh.enabled: true
+   * </pre>
+   *
+   * @return HdfsDelegationTokenRefresher
+   */
+  @Bean
+  public HdfsDelegationTokenRefresher getDelegationTokenRefresher() {
+    if (!hdfsStorage.isConfigured()) {
+      log.debug(
+          "Hdfs storage is not configured, ignoring HdfsDelegationTokenRefresher bean creation");
+      return null;
+    }
+    String refreshEnabled =
+        hdfsStorage.getProperties().getOrDefault(HDFS_TOKEN_REFRESH_ENABLED, "false");
```

Comment:
Should we enable delegation token refresh by default if HDFS storage is configured? Because token needs to be refreshed if HDFS is accessed.

Reply:
One issue I see with OSS codebase is token is not available by default. The HADOOP_TOKEN_FILE_LOCATION can be presented token is not in UGI. Off course this can be enabled using config. For li internal repo all of these are enabled by default using deployment.

Reply:
But again OSS perspective there could other option as well Kerberos etc, might not rely on UGI approach like we rely on UGI and hadoop token file internally at Li. I think should be okay to have it disabled by default.

Reply:
It seems like it would make sense to have token refresh disabled by default. Security is disabled by default in OSS Hadoop.